Add a WAM regression test with the MULTI_GASES and MOLECULAR_DIFFUSION options on

XiaqiongZhou-NOAA opened the pull request:
## Description

Add a WAM regression test with the MULTI_GASES and MOLECULAR_DIFFUSION options on


## Testing



## Dependencies
https://github.com/ufs-community/ufs-weather-model/pull/714
https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/120
https://github.com/NCAR/ccpp-physics/pull/712

SajalKar-NOAA committed:
VLab issue # 76949. Update FV3-WAM from IPD to CCPP

SajalKar-NOAA committed:
Update FV3-WAM from IPD to CCPP

Commit:
Changed machine-dependency for Orion.

Commit:
consistent commit for dadhmhj.01a for FV3

HenryJuang-NOAA committed:
make MOLECULAR_DIFFUSION on without MULTI_GASES

HenryJuang-NOAA committed:
fix ccpp_build.sh due to SION

HenryJuang-NOAA committed:
update for dadhmhj.02

Commit:
Added IDEA-physics files from soon-to-be-operational WAM-GSM.

HenryJuang-NOAA committed:
update .gitmodules to have submodules point to udhmhj.02

HenryJuang-NOAA committed:
change UV molecular diffusion without extra interpolation

HenryJuang-NOAA committed:
modify molecular diffusion for u and v without extra inter[pation effect

HenryJuang-NOAA committed:
molecular diffusion on cell edge D grid + non-sli[ BC for update_dz_c and d

Commit:
All IDEA-physics routines working with IPD

HenryJuang-NOAA committed:
Update molecular diffusion on D grid

Commit:
Added a new file

HenryJuang-NOAA committed:
Merge branch 'udhmhj.03' of https://github.com/HenryJuang-NOAA/fv3atm into udhmhj.04
fix conflicy

HenryJuang-NOAA committed:
merged *cub* and ccpp/physics with dadsk.03

HenryJuang-NOAA committed:
Merge branch 'dadsk.03' of https://github.com/akubaryk/fv3atm into udhmhj.04
merge with conflict fixed.

HenryJuang-NOAA committed:
update to udhmhj.04

HenryJuang-NOAA committed:
Merge branch 'develop' into udhmhj.04
commit with conflict fixed.
Conflicts:
	.gitmodules
	atmos_cubed_sphere
	ccpp/build_ccpp.sh
	ccpp/physics
	ccpp/set_compilers.sh
	ccpp/suites/suite_FV3_GFS_2017_fv3wam.xml
	gfsphysics/GFS_layer/GFS_physics_driver.F90
	gfsphysics/GFS_layer/GFS_typedefs.F90
	gfsphysics/physics/get_prs.f

HenryJuang-NOAA committed:
incremental commit

HenryJuang-NOAA committed:
use Cmake and can be compiled in orion with all options MG MD IP

HenryJuang-NOAA committed:
add md_reflev option 0 and some fixes for typedefs

HenryJuang-NOAA committed:
add first DEEP_ATMOS_DYNAMICS

HenryJuang-NOAA committed:
fix spfox for ntoz, nto, and nto2.

HenryJuang-NOAA committed:
change spfox to spox for consisyency with sphum.

HenryJuang-NOAA committed:
modify molecular diffusion and tau_w/

XiaqiongZhou-NOAA committed:
Point to new personal branch

XiaqiongZhou-NOAA committed:
Merge with NOAA-EMC

XiaqiongZhou-NOAA committed:
Update dycore

XiaqiongZhou-NOAA committed:
Merge remote-tracking branch 'NOAA-EMC/develop' into udhmhj

XiaqiongZhou-NOAA committed:
Update submodules to new versions

XiaqiongZhou-NOAA committed:
Merge remote-tracking branch 'NOAA-EMC/develop' into udhmhj6

XiaqiongZhou-NOAA committed:
correct merge errors

XiaqiongZhou-NOAA committed:
Correct merge errors

XiaqiongZhou-NOAA committed:
Update CCPP suites for the MULTI_GASES RT

XiaqiongZhou-NOAA committed:
Update atmos_cubed_sphere to a new version

HenryJuang-NOAA committed:
add md_tadj_layers and fix o3mr with spo3

XiaqiongZhou-NOAA committed:
Merge HenryJuang's branch

XiaqiongZhou-NOAA committed:
update submodules

XiaqiongZhou-NOAA committed:
Merge remote-tracking branch 'NOAA-EMC/develop' into udhmhj6

XiaqiongZhou-NOAA committed:
Update atmos_cubed_sphere and ccpp/physics submodules

XiaqiongZhou-NOAA committed:
Merge NOAA-EMC/develop to udhmhj6

XiaqiongZhou-NOAA committed:
Change file permissions

XiaqiongZhou-NOAA committed:
Change file permisssions and update submodules

XiaqiongZhou-NOAA committed:
Delete the suites are not used

DusanJovic-NOAA commented:
Why are all these `idea_*.f` files added to `gfsphysics/physics/` directory in this repo, instead of `ccpp/physics`?

climbfuji commented on ccpp/data/GFS_typedefs.F90:
```diff
@@ -4755,6 +4789,31 @@ subroutine control_initialize (Model, nlunit, fn_nml, me, master, &
+    ! setting them here can lead to memory corruption that is hard to detect.
+!--- stored in wam_f107_kp module
+    
+    lsidea            = .false.
```
This is not how CCPP works, unfortunately. You cannot set module variables that reside in some module in ccpp-physics in `GFS_typedefs.F90` directly. You should create metadata tables for these variables and add them to the variable definition tables, then set them in the physics init phase.

I think Thompson microphysics is a good example to consult.

XiaqiongZhou-NOAA commented:
cced to @HenryJuang-NOAA 
Idealized physics is not working within CCPP. Henry, should it be deleted from gfsphysics/physics and move to ccpp/physics？

climbfuji reviewed (changes requested):
I see a number of files being added in the old IPD directory structure. How are these being called? There are no drivers anymore. i think this code should be removed from this PR and added to CCPP physics, if desired.

DusanJovic-NOAA commented:
> cced to @HenryJuang-NOAA
> Idealized physics is not working within CCPP. Henry, should it be deleted from gfsphysics/physics and move to ccpp/physics？

I do not even see where these files are compiled. Are they used?

yangfanglin commented:
The WAM physics development is on hold.  These files were coupled to the
dynamics through IPD.  Work needs to be done to move them under CCPP, with
testing and evaluation.  We are in the process of hiring a new WAM
developer.

So the question is where should these files be stored for now ? They can be
kept in a personal branch for the time being.

Fanglin

On Tue, Jul 27, 2021 at 4:05 PM Dusan Jovic ***@***.***>
wrote:

> cced to @HenryJuang-NOAA <https://github.com/HenryJuang-NOAA>
> Idealized physics is not working within CCPP. Henry, should it be deleted
> from gfsphysics/physics and move to ccpp/physics？
>
> I do not even see where these files are compiled. Are they used?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/fv3atm/pull/354#issuecomment-887798416>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKY5N2I3WZNQQMHPHABV363TZ4GQHANCNFSM5BCN5JIA>
> .
>


-- 
*Fanglin Yang, Ph.D.*
*Chief, Model Physics Group*
*Modeling and Data Assimilation Branch*

*NOAA/NWS/NCEP Environmental Modeling Center*

*https://www.emc.ncep.noaa.gov/gmb/wx24fy/fyang/
<https://www.emc.ncep.noaa.gov/gmb/wx24fy/fyang/>*

climbfuji commented:
> The WAM physics development is on hold. These files were coupled to the dynamics through IPD. Work needs to be done to move them under CCPP, with testing and evaluation. We are in the process of hiring a new WAM developer. So the question is where should these files be stored for now ? They can be kept in a personal branch for the time being. Fanglin
> […](#)

Is the purpose of these PRs to store this IPD code for future work? If so, then it should go into a branch in my opinion ...

yangfanglin commented:
> > The WAM physics development is on hold. These files were coupled to the dynamics through IPD. Work needs to be done to move them under CCPP, with testing and evaluation. We are in the process of hiring a new WAM developer. So the question is where should these files be stored for now ? They can be kept in a personal branch for the time being. Fanglin
> > […](#)
> 
> Is the purpose of these PRs to store this IPD code for future work? If so, then it should go into a branch in my opinion ...

@XiaqiongZhou-NOAA @HenryJuang-NOAA  please confirm if this is the case. If yes, Kate you can create a branch of your own to save them for future use.

yangfanglin commented:
@SajalKar-NOAA   add Sajal to the discussion for clarification as well.

yangfanglin commented:
Kate, do you need the IDEA physics for creating the new WAM MULTI_GAS /Molecular diffusion regression test ?   If not, I think you can save the IDEA physics under gfsphysics to a branch, and commit the rest from this PR.

XiaqiongZhou-NOAA commented:
IDEA physics is not used. I can make a copy of the current version and delete this part for merge.

HenryJuang-NOAA commented:
The purpose of this PR is for WAM dynamics including multi-gases and
molecular-diffusion working together ready for WAM physics. I agree to save
IDEA_*  in a personal branch, and wait for the next PR when it is in CCPP
ready later for future PR.


On Tue, Jul 27, 2021 at 4:43 PM Fanglin Yang ***@***.***>
wrote:

> The WAM physics development is on hold. These files were coupled to the
> dynamics through IPD. Work needs to be done to move them under CCPP, with
> testing and evaluation. We are in the process of hiring a new WAM
> developer. So the question is where should these files be stored for now ?
> They can be kept in a personal branch for the time being. Fanglin
> … <#m_-8348961299337377182_>
>
> Is the purpose of these PRs to store this IPD code for future work? If so,
> then it should go into a branch in my opinion ...
>
> @XiaqiongZhou-NOAA <https://github.com/XiaqiongZhou-NOAA> @HenryJuang-NOAA
> <https://github.com/HenryJuang-NOAA> please confirm if this is the case.
> If yes, Kate you can create a branch of your own to save them for future
> use.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/fv3atm/pull/354#issuecomment-887821115>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRWWEMHXFTRPKCSZTHIB7LTZ4K6LANCNFSM5BCN5JIA>
> .
>

XiaqiongZhou-NOAA committed:
Delete the part related to the IDEA physics. It will be added in CCPP in the future

XiaqiongZhou-NOAA committed:
Delete the part related to the IDEA physics. It will be added in CCPP in the future

XiaqiongZhou-NOAA commented:
I have deleted gfsphysics/physics and  deleted some parts related to the IDEA physics in ccpp/data/GFS_typedefs.F90. T The undeleted version is in my branch: https://github.com/XiaqiongZhou-NOAA/ufs-weather-model/tree/udhmhj7 for the future development.

SajalKar-NOAA commented:
I repeat; why do you have to delete the IDEA physics bit?


Dr. Sajal K. Kar
IMSG at NOAA/NWS/NCEP/EMC
5830 University Research Ct., Rm. 2176
College Park, MD 20740
***@***.***
301-683-3782


On Wed, Jul 28, 2021 at 9:45 AM XiaqiongZhou-NOAA ***@***.***>
wrote:

> I have deleted gfsphysics/physics and deleted some parts related to the
> IDEA physics in ccpp/data/GFS_typedefs.F90. T The undeleted version is in
> my branch:
> https://github.com/XiaqiongZhou-NOAA/ufs-weather-model/tree/udhmhj7 for
> the future development.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/fv3atm/pull/354#issuecomment-888321630>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANNLCTDD3OM2ESGQRJOE7YLT2ACXZANCNFSM5BCN5JIA>
> .
>

XiaqiongZhou-NOAA commented:
> I repeat; why do you have to delete the IDEA physics bit? Dr. Sajal K. Kar IMSG at NOAA/NWS/NCEP/EMC 5830 University Research Ct., Rm. 2176 College Park, MD 20740 ***@***.*** 301-683-3782
> […](#)
> On Wed, Jul 28, 2021 at 9:45 AM XiaqiongZhou-NOAA ***@***.***> wrote: I have deleted gfsphysics/physics and deleted some parts related to the IDEA physics in ccpp/data/GFS_typedefs.F90. T The undeleted version is in my branch: https://github.com/XiaqiongZhou-NOAA/ufs-weather-model/tree/udhmhj7 for the future development. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <[#354 (comment)](https://github.com/NOAA-EMC/fv3atm/pull/354#issuecomment-888321630)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ANNLCTDD3OM2ESGQRJOE7YLT2ACXZANCNFSM5BCN5JIA> .

Sajia: I committed twice to finish the changes that I want to make. I am not sure which part in GFS_typdefs.F90 is OK for the future CCPP. Please let me know which part should be remained.

SajalKar-NOAA commented:
Kate,
Thank you for your reply. I don't have time for this. So, go ahead with
your work.
Sajal

On Wednesday, July 28, 2021, XiaqiongZhou-NOAA ***@***.***>
wrote:

> I repeat; why do you have to delete the IDEA physics bit? Dr. Sajal K. Kar
> IMSG at NOAA/NWS/NCEP/EMC 5830 University Research Ct., Rm. 2176 College
> Park, MD 20740
> <https://www.google.com/maps/search/5830+University+Research+Ct.,+Rm.+2176+College+Park,+MD+20740?entry=gmail&source=g>
> *@*.*** 301-683-3782
> … <#m_8376816571670609758_>
> On Wed, Jul 28, 2021 at 9:45 AM XiaqiongZhou-NOAA *@*.***> wrote: I have
> deleted gfsphysics/physics and deleted some parts related to the IDEA
> physics in ccpp/data/GFS_typedefs.F90. T The undeleted version is in my
> branch: https://github.com/XiaqiongZhou-NOAA/ufs-weather-
> model/tree/udhmhj7 for the future development. — You are receiving this
> because you were mentioned. Reply to this email directly, view it on GitHub
> <#354 (comment)
> <https://github.com/NOAA-EMC/fv3atm/pull/354#issuecomment-888321630>>, or
> unsubscribe https://github.com/notifications/unsubscribe-auth/
> ANNLCTDD3OM2ESGQRJOE7YLT2ACXZANCNFSM5BCN5JIA .
>
> Sajia: I committed twice to finish the changes that I want to make. I am
> not sure which part in GFS_typdefs.F90 is OK for the future CCPP. Please
> let me know which part should be remained.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/fv3atm/pull/354#issuecomment-888373450>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANNLCTHX5J27L7R7LM2QPRDT2AKINANCNFSM5BCN5JIA>
> .
>


-- 
Dr. Sajal K. Kar
IMSG at NOAA/NWS/NCEP/EMC
5830 University Research Ct., Rm. 2176
College Park, MD 20740
***@***.***
301-683-3782

yangfanglin commented:
Kate,

Please move ahead with your PR without including IDEA physics in GFS_typdefs.F90 etc since it is still sitting in the  IPD.   The connections need to be re-established after IDEA physics is moved to CCPP.

SajalKar-NOAA commented:
It's only in IPD at this stage. Thank you, Fanglin.

Dr. Sajal K. Kar
IMSG at NOAA/NWS/NCEP/EMC
5830 University Research Ct., Rm. 2176
College Park, MD 20740
***@***.***
301-683-3782


On Wed, Jul 28, 2021 at 11:17 AM Fanglin Yang ***@***.***>
wrote:

> Kate,
>
> Please move ahead with your PR without including IDEA physics in
> GFS_typdefs.F90 etc since it is still sitting in the IPD. The connections
> need to be re-established after IDEA physics is moved to CCPP.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/fv3atm/pull/354#issuecomment-888395473>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANNLCTFLLLWPJQF6OUKYITDT2ANOVANCNFSM5BCN5JIA>
> .
>

junwang-noaa commented on ccpp/data/GFS_typedefs.F90:
```diff
@@ -3112,8 +3111,8 @@ subroutine control_initialize (Model, nlunit, fn_nml, me, master, &
     integer              :: levr           = -99             !< number of vertical levels for radiation calculations
-    integer              :: nfxr           = 39+6            !< second dimension of input/output array fluxr
-    logical              :: iaerclm        = .false.         !< flag for initializing aero data
+    integer              :: nfxr           = 39+6            !< second dimension of input/output array fluxr   
```
Please remove the spaces in the two lines

junwang-noaa commented on ccpp/data/GFS_typedefs.F90:
```diff
@@ -2256,7 +2255,7 @@ module GFS_typedefs
 ! GFS_statein_type%create
 !------------------------
-  subroutine statein_create (Statein, IM, Model)
+  subroutine statein_create (Statein, IM, Model) 
```
remove space at the end of the line

junwang-noaa commented on ccpp/data/GFS_typedefs.F90:
```diff
@@ -2082,7 +2081,7 @@ module GFS_typedefs
     real (kind=kind_phys), pointer      :: vegf1d(:)          => null()  !<
     real (kind=kind_phys)               :: lndp_vgf                      !<
-
+  
```
Please revert the change

XiaqiongZhou-NOAA committed:
Delete unneeded whitespace

XiaqiongZhou-NOAA committed:
Delete more whitespace

XiaqiongZhou-NOAA committed:
Update GFS_typedefs.F90

XiaqiongZhou-NOAA committed:
Remove the molecular diffusion compiling option

XiaqiongZhou-NOAA committed:
Update atmos_cubed_sphere and ccpp/physics submodules

XiaqiongZhou-NOAA committed:
Merge remote tracking NOAA-EMC/develop to udhmhj6

XiaqiongZhou-NOAA committed:
Update framework submodule

XiaqiongZhou-NOAA committed:
Merge remote-tracking branch 'NOAA-EMC/develop' into udhmhj6

XiaqiongZhou-NOAA committed:
update the ccpp/physics and atmos_cubed-sphere submodules

XiaqiongZhou-NOAA committed:
Merge remote-tracking branch 'NOAA-EMC/develop' into udhmhj6

XiaqiongZhou-NOAA committed:
Update the ccpp/framework submodule

XiaqiongZhou-NOAA committed:
Update ccpp/physics

XiaqiongZhou-NOAA committed:
Merge remote-tracking branch 'NOAA-EMC/develop' into udhmhj6

XiaqiongZhou-NOAA committed:
Update the atmos_cubed_sphere submodule

XiaqiongZhou-NOAA committed:
Update atmos_cubed_sphere

XiaqiongZhou-NOAA committed:
Update the submodule atmos_cubed_sphere to a new version

XiaqiongZhou-NOAA committed:
Merge remote-tracking branch 'NOAA-EMC/develop' into udhmhj6

XiaqiongZhou-NOAA committed:
Update ccpp/physics submodule

XiaqiongZhou-NOAA committed:
Merge remote-tracking branch 'NOAA-EMC/develop' into udhmhj6

XiaqiongZhou-NOAA committed:
Update ccpp-physiscs

XiaqiongZhou-NOAA committed:
Merge remote-tracking branch NOAA-EMC/develop to udhmhj6

XiaqiongZhou-NOAA committed:
Point submodules to the public version
